src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -806,171 +996,193 @@ class C
+                var model = compilation.GetSemanticModel(tree);
+                var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+                var x = nodes.OfType<VariableDeclaratorSyntax>().First();
+
+                Assert.Equal("<tuple: System.Int32 a, System.Int32 b, System.Int32 c, System.Int32 d, System.Int32 e, System.Int32 f, System.Int32 g, "
+                    + "System.String h, System.Int32 i, System.Int32 j, System.Int32 k, System.Int32 l> x",
+                    model.GetDeclaredSymbol(x).ToTestDisplayString());
+            };
+
+            var verifier = CompileAndVerify(source, expectedOutput: @"1 2 3 4 5 6 7 Alice 2 3 4 5", additionalRefs: new[] { MscorlibRef }, sourceSymbolValidator: validator);
+            verifier.VerifyDiagnostics();
         }
 
         [Fact]
-        public void MethodReturnsValueTuple()
+        public void LongTupleCreation()
         {
-            var source = @"
+            var source = trivial2uple + trivial3uple + trivalRemainingTuples + @"
 class C
 {
     static void Main()
     {
-        System.Console.WriteLine(M().ToString());
-    }
-
-    static (int, string) M()
-    {
-        return (1, ""hello"");
+        var x = (1, 2, 3, 4, 5, 6, 7, ""Alice"", 2, 3, 4, 5, 6, 7, ""Bob"", 2, 3);
+        System.Console.WriteLine($""{x.Item1} {x.Item2} {x.Item3} {x.Item4} {x.Item5} {x.Item6} {x.Item7} {x.Item8} {x.Item9} {x.Item10} {x.Item11} {x.Item12} {x.Item13} {x.Item14} {x.Item15} {x.Item16} {x.Item17}"");
     }
 }
-" + trivial2uple;
+";
 
-            var comp = CompileAndVerify(source, expectedOutput: @"{1, hello}");
+            Action<ModuleSymbol> validator = module =>
+            {
+                var sourceModule = (SourceModuleSymbol)module;
+                var compilation = sourceModule.DeclaringCompilation;
+                var tree = compilation.SyntaxTrees.First();
+                var model = compilation.GetSemanticModel(tree);
+                var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+                var node = nodes.OfType<TupleExpressionSyntax>().Single();
+
+                Assert.Equal("<tuple: System.Int32 Item1, System.Int32 Item2, System.Int32 Item3, System.Int32 Item4, System.Int32 Item5, System.Int32 Item6, System.Int32 Item7, "
+                     + "System.String Item8, System.Int32 Item9, System.Int32 Item10, System.Int32 Item11, System.Int32 Item12, System.Int32 Item13, System.Int32 Item14, "
+                     + "System.String Item15, System.Int32 Item16, System.Int32 Item17>",
+                     model.GetTypeInfo(node).Type.ToTestDisplayString());
+            };
+
+            var verifier = CompileAndVerify(source, expectedOutput: @"1 2 3 4 5 6 7 Alice 2 3 4 5 6 7 Bob 2 3", additionalRefs: new[] { MscorlibRef }, sourceSymbolValidator: validator);
+            verifier.VerifyDiagnostics();
         }
 
         [Fact]
-        public void Tuple2To7Members()
+        public void LongTupleCreationWithNames()
```

Comment:
I think we also need a test confirming the order of argument evaluation. It should be left to right. The easiest way is to execute something like:
```C#
 var t = (a1: PrintAndReturn(1), a2: PrintAndReturn(2),... , a18: PrintAndReturn(18)); 
```